CONTRIBUTING.md

```diff
@@ -0,0 +1,80 @@
+
+## Test ReactOS
+
+### Localize bugs
+
+By localizing bugs, developers can identify what causes the bug and what it affects. There are a variety of methods to [debug] ReactOS while testing it. After identifying a bug, check if it is already known about by searching [JIRA] and adding any additional information to the report. If you think that it is an unidentified bug, consider [filing a bug report].
+
+### Fix bugs
+
+Instead of looking for bugs, you can also try to fix a few that are already listed in JIRA. Fixing bugs requires a lot more skill than simply searching for them, and can be time consuming.
+
+### Write tests
+
+Tests are used to check the functionality and correctness of APIs on ReactOS compared to Windows implementations. There are also some unit tests that you could help ReactOS pass, which can be found [here][testman].
+
+## Implement new things
+
+Considering ReactOS is alpha quality software, there is a lot of [missing functionality] that Windows operating systems have. Before starting a project to implement something, find out if another person is working on the same thing. If you find that someone is already working on it, ask if any assistance is needed for what specifically is being worked on or a related project. Plenty of times a person will start to implement something and never finish before moving to something else. Make sure you stay committed to what you are going to implement, and do not be afraid to ask for assistance if you need help with something.
+
+## Contribute
+
+Your contribution can be of numerous forms. We currently accept two ways to contribute - Pull Requests and Patches.
+
+### Pull Requests
+
+Since our [migration to GitHub] we gladly accept [Pull Requests]. Pull requests let you tell others about changes you've pushed to a repository on GitHub. Once a pull request is opened, you can discuss and review the potential changes with collaborators and add follow-up commits before the changes are merged into the repository. Pull request is a prefered way to submit your work - it makes reviewing and merging your contribution much easier.
+
+### Patches
+
+A [patch] is a set of changes to existing source code. The changes in a patch can be merged into existing source code. This process is referred to as applying a patch (to source code). Which changes a patch contains and the way the patch is structured can have significant impact on the consequences that can happen from applying the patch. 
+
+See [Submitting Patches] for details.
+
+### Rules and Recomendations
+Regardless of your way of contributing to ReactOS we have some rules and recomendations:
+
+- Use your __full real name__ and __real email__. We don't accept anonymous contributions!
+- Ensure your contribution is properly described. Include the relevant issue number if applicable.
+- Put only related changes. It will make reviewing easier as the reviewer needs to recall less information about the existing source code that is changed.
```

Comment:
"recall" or "remember"?

Reply:
`recall` is better in this context IMO. You open the changed code to _recall_ details, but you don't need to _remember_ it before you review.